plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/actions/layout/PasteImageAction.java

```diff
@@ -0,0 +1,179 @@
+package org.eclipse.sirius.diagram.ui.tools.internal.actions.layout;
+
+import java.awt.Toolkit;
+import java.awt.datatransfer.FlavorEvent;
+import java.awt.datatransfer.FlavorListener;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.text.MessageFormat;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.emf.common.command.CompoundCommand;
+import org.eclipse.emf.transaction.TransactionalEditingDomain;
+import org.eclipse.emf.transaction.util.TransactionUtil;
+import org.eclipse.jface.action.Action;
+import org.eclipse.sirius.business.api.image.ImageManager;
+import org.eclipse.sirius.business.api.image.ImageManager.CreateFileFunc;
+import org.eclipse.sirius.business.api.image.ImageManagerProvider;
+import org.eclipse.sirius.diagram.ui.business.api.image.WorkspaceImageHelper;
+import org.eclipse.sirius.diagram.ui.provider.DiagramUIPlugin;
+import org.eclipse.sirius.diagram.ui.provider.Messages;
+import org.eclipse.sirius.diagram.ui.tools.api.image.DiagramImagesPath;
+import org.eclipse.sirius.diagram.ui.tools.api.ui.actions.ActionIds;
+import org.eclipse.sirius.diagram.ui.tools.internal.actions.style.SetStyleToWorkspaceImageAction;
+import org.eclipse.sirius.tools.api.SiriusPlugin;
+import org.eclipse.sirius.viewpoint.BasicLabelStyle;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.dnd.Clipboard;
+import org.eclipse.swt.dnd.ImageTransfer;
+import org.eclipse.swt.graphics.ImageData;
+import org.eclipse.swt.graphics.ImageLoader;
+import org.eclipse.swt.widgets.Display;
+
+/**
+ * This action provides past image from the clipboard.
+ * 
+ * This action provides past image from the clipboard on Sirius elements that can have workspace image style. This
+ * action create image in image forder of project from clipboard and then set as workspace image on all selected nodes.
+ * 
+ * @author Séraphin Costa
+ */
+public class PasteImageAction extends Action {
+
+    // There are no clipboard listener in SWT, so we need to use the AWT clipboard listener.
+    java.awt.datatransfer.Clipboard awtClipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
+
+    class ClipboardListener implements FlavorListener {
+        @Override
+        public void flavorsChanged(FlavorEvent e) {
+            // This is an AWT event, not an SWT event,
+            // so we are not in the thread UI and we need to execute this code in thread UI.
+            Display.getDefault().syncExec(() -> {
+                updateState();
+            });
+        }
+    }
+
+    /**
+     * Default constructor.
+     */
+    public PasteImageAction() {
+        super();
+
+        setId(ActionIds.PASTE_IMAGE);
+        setText(Messages.PasteImageAction_text);
+        setToolTipText(Messages.PasteImageAction_toolTipText);
+        setAccelerator(SWT.CTRL | SWT.ALT | 'V');
+
```

Comment:
This new action should be documented in the user documentation.
Are you sure about the keyboard shortcut? It does not seem to work on my environment (linux)